### ISSUE TITLE
Allow providing the max batch size as a cli arg

### ISSUE THREAD

**Issue by brianmcgee** [numtide/treefmt#334]
**Is your feature request related to a problem? Please describe.**

Currently, all batching is hard coded to a maximum of `1024`. In certain circumstances, it might be beneficial to specify smaller batch sizes to improve concurrency. 

**Describe the solution you'd like**

Add a cli arg, `--batch-size` which controls the max batch size.

**Comment by brianmcgee** [comment]
Yeah, worth revisiting this. I'll have some time for treefmt at the end of this week. It would be nice to get this in before releasing #571

**Comment by jfly** [comment]
I chatted with @brianmcgee about this: gonna hack on this a bit, shouldn't take too long.
